packages/core/src/fhir-to-cda/cda-templates/components/medications.ts

```diff
@@ -151,77 +144,115 @@ function createAugmentedMedicationStatements(
   referenceId: string
-): SubstanceAdministationEntry[] {
+): SubstanceAdministationEntry {
   const manufacturedMaterialCode = buildCodeCvFromCodeableConcept(
     statement.medication?.code,
     referenceId
   );
-  return [
-    {
-      substanceAdministration: {
-        [_classCodeAttribute]: "SBADM",
-        [_moodCodeAttribute]: "INT",
-        templateId: buildInstanceIdentifier({
-          root: statement.typeOid,
-          extension: extensionValue2014,
-        }),
-        id: buildInstanceIdentifier({
-          root: placeholderOrgOid,
-          extension: statement.resource.id,
-        }),
-        statusCode: {
-          [_codeAttribute]: statement.resource.status,
-        },
-        effectiveTime: {
-          [_xsiTypeAttribute]: "IVL_TS",
-          low: withoutNullFlavorObject(
-            formatDateToCdaTimestamp(statement.resource.effectivePeriod?.start),
-            _valueAttribute
-          ),
-          high: withoutNullFlavorObject(
-            formatDateToCdaTimestamp(statement.resource.effectivePeriod?.end),
-            _valueAttribute
-          ),
-        },
-        consumable: {
-          [_typeCodeAttribute]: "CSM",
-          manufacturedProduct: {
-            templateId: buildInstanceIdentifier({
-              root: "2.16.840.1.113883.10.20.22.4.23",
-              extension: "2014-06-09",
-            }),
-            manufacturedMaterial: {
-              code: manufacturedMaterialCode,
-            },
+  return {
+    substanceAdministration: {
+      [_classCodeAttribute]: "SBADM",
+      [_moodCodeAttribute]: "INT",
+      templateId: buildInstanceIdentifier({
+        root: statement.typeOid,
+        extension: extensionValue2014,
+      }),
+      id: buildInstanceIdentifier({
+        root: placeholderOrgOid,
+        extension: statement.resource.id,
+      }),
+      statusCode: {
+        [_codeAttribute]: statement.resource.status,
+      },
+      effectiveTime: {
+        [_xsiTypeAttribute]: "IVL_TS",
+        low: withoutNullFlavorObject(
+          formatDateToCdaTimestamp(statement.resource.effectivePeriod?.start),
+          _valueAttribute
+        ),
+        high: withoutNullFlavorObject(
+          formatDateToCdaTimestamp(statement.resource.effectivePeriod?.end),
+          _valueAttribute
+        ),
+      },
+      consumable: {
+        [_typeCodeAttribute]: "CSM",
+        manufacturedProduct: {
+          templateId: buildInstanceIdentifier({
+            root: oids.medicationInformation,
+            extension: extensionValue2014,
+          }),
+          manufacturedMaterial: {
+            code: manufacturedMaterialCode,
           },
         },
       },
     },
-  ];
+  };
+}
+
+function buildMedicationCode(code: CdaCodeCe | undefined): string {
+  if (code?._code) {
+    if (code?._codeSystem) return `${code._code} - ${code._codeSystem}`;
+    if (code?._codeSystemName) return `${code._code} - ${code._codeSystemName}`;
```

Comment:
nit: `code` at this point can't be undefined, so no need for `?`

Reply:
Ah, good catch